src/main/resources/google_checks.xml

```diff
@@ -78,12 +78,19 @@
                value="LITERAL_DO, LITERAL_ELSE, LITERAL_FOR, LITERAL_IF, LITERAL_WHILE"/>
     </module>
     <module name="LeftCurly">
+      <property name="id" value="LeftCurlyEol"/>
       <property name="tokens"
                value="ANNOTATION_DEF, CLASS_DEF, CTOR_DEF, ENUM_CONSTANT_DEF, ENUM_DEF,
-                    INTERFACE_DEF, LAMBDA, LITERAL_CASE, LITERAL_CATCH, LITERAL_DEFAULT,
-                    LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR, LITERAL_IF,
-                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_WHILE, METHOD_DEF,
-                    OBJBLOCK, STATIC_INIT, RECORD_DEF, COMPACT_CTOR_DEF"/>
+                    INTERFACE_DEF, LAMBDA, METHOD_DEF, OBJBLOCK, LITERAL_SWITCH,
+                    STATIC_INIT, RECORD_DEF, COMPACT_CTOR_DEF, LITERAL_CATCH, LITERAL_TRY,
+                    LITERAL_WHILE, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, LITERAL_FOR,
+                    LITERAL_IF, LITERAL_SYNCHRONIZED"/>
+    </module>
+    <module name="LeftCurly">
+      <property name="id" value="LeftCurlyNl"/>
+      <property name="option" value="nl"/>
+      <property name="tokens"
+                value="LITERAL_CASE, LITERAL_DEFAULT" />
```

Comment:
Rule says that a block of statements can appear anywhere in the code and opening curly braces of such blocks are preceded by line break. 

> Exception: In places where these rules allow a single statement ending with a semicolon (;), a block of statements can appear, and the opening brace of this block is preceded by a line break. Blocks like these are typically introduced to limit the scope of local variables, for example inside switch statements.

Here by the `block of statements` means a block of code, something like:

```
{
  int x = 0;
  foo(x++);
}
```

These block of code can appear anywhere, inside an if statement, method, switch statement, etc. Opening braces of such blocks needs to be preceded with a line break.

Here, you have added only LITERAL_CASE & LITERAL_DEFAULT in the tokens, by this way the rule will only be true for switch statements, but we need to follow this rule everywhere else too. 

Your approach is good but the tokens you have used are incorrect. Maybe `INSTANCE_INIT` will do the work.

Reply:
I will solve this issue later, you may close this PR.

Reply:
nvm, you can continue your work as maintainer suggested here: https://github.com/checkstyle/checkstyle/pull/15335#issuecomment-2241740935.

Reply:
@Zopsss , you are first reviewer of this PR, I will look only after you finish.

Reply:
@kvlohith1 ping

Pls go through my comment: https://github.com/checkstyle/checkstyle/pull/15335#discussion_r1685715555. This is a difficult task, if you're a beginner I would suggest you to work on [good-first issues](https://github.com/checkstyle/checkstyle/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) to get some experience. But if you still want to work on this issue, we're here to help you. Let us know what you decide :)

Reply:
Hi @Zopsss, 
I am not a beginner (albeit a beginner to Checkstyle), I would like to still work on it if thats alright.

Reply:
alrighty then let's solve this issue! You can go through my this comment: https://github.com/checkstyle/checkstyle/pull/15335#discussion_r1685715555. We need to find out a way to detect such blocks. Play with our CLI and generate ASTs to find a way to detect such blocks, it would be good if we're able to achieve this by modifying our google config only. If we cannot achieve this only by modifying our config then we would need to make modifications in check's implementation. If you face some problems or have any doubts, don't hesitate to ask us :)

Reply:
Apologies, I have been working on this, feels a little too complicated.

Reply:
Did you tried generating ASTs and found any patterns? If yes then please share